general_stats_generator.py

```diff
@@ -0,0 +1,33 @@
+from functools import reduce
+from services import DataServices
+
+class GeneralStatsCreator(DataServices):
+    tittles = ['Total Trip Count', 'Max Trip Time', 'Total Bike Count',
+               'Unprocessed Rows Count']  # for output file header
+    stats = []
+
+    def __init__(self, obj):
+        self.all_trips = obj.all_trips
+        self.unprocessed_count = obj.unprocessed_count
+        self.gather_stats()
+
+    def gather_stats(self):
+        trips_amount = self.get_general_trips_amount(self.all_trips)
```

Comment:
as you have access to all trips from this class, why the length can't be calculated from here?

Reply:
I left this method in parent class since in usage_stats_generator.py it is also used, and I thought that was a good idea